k-distribution/include/z3/exp.smt2

```diff
@@ -0,0 +1,69 @@
+(set-option :auto-config false)
+(set-option :smt.mbqi false)
+;(set-option :smt.mbqi.max_iterations 15)
+
+(declare-fun pow256 () Int)
+;(assert (= pow256 115792089237316195423570985008687907853269984665640564039457584007913129639936))
+(assert (>= pow256 115792089237316195423570985008687907853269984665640564039457584007913129639936))
+(assert (<= pow256 115792089237316195423570985008687907853269984665640564039457584007913129639936))
+
+;
+; expFunc
+;
+
+(declare-fun expFunc (Int Int) Int)
+
+(assert (forall ((x1 Int) (y1 Int) (x2 Int) (y2 Int))
+  (!
+    (=> (and (<= x1 x2) (<= y1 y2))
+      (<= (expFunc x1 y1) (expFunc x2 y2))
+    )
+    :pattern ((expFunc x1 y1) (expFunc x2 y2))
+  )
+))
+
+(assert (forall ((x Int) (y Int)) (! (=> (>= y 1) (>= (expFunc x y) x      )) :pattern ((expFunc x y)))))
+(assert (forall ((x Int) (y Int)) (! (=> (>= y 2) (>= (expFunc x y) (* x x))) :pattern ((expFunc x y)))))
+
+;
+; smt_rpow
+;
+
+(declare-fun smt_rpow (Int Int Int Int) Int)
+(assert (forall ((z Int) (x Int) (y Int) (b Int)) (! (=> (= y 1) (= (smt_rpow z x y b) (div (+ (* z x) (div b 2)) b))) :pattern ((smt_rpow z x y b)))))
```

Comment:
I do not remember what this was for. It seems not consistent with others. You can test if this is still needed.